File: 171. Excel Sheet Column Number/171. Excel Sheet Column Number.js
/**
 * @param {string} columnTitle
 * @return {number}
 */
var titleToNumber = function(columnTitle) {
    let result = 0;
    for (let c of columnTitle) {
        result = result * 26 + (c.charCodeAt(0) - 64);
    }
    return result;
};

File: 171. Excel Sheet Column Number/171. Excel Sheet Column Number.py
class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        a = [None] * 28
        j = 1
        for i in alpha:
            a[j] = i
            j += 1
        if len(columnTitle) == 1:
            return a.index(columnTitle)
        else:
            out = 0
            for x in columnTitle:
                out = out * 26 + a.index(x)
            return out
            